src/Microsoft.AspNetCore.DataProtection/DataProtectionBuilderExtensions.cs

```diff
@@ -424,61 +474,73 @@ public DataProtectionConfiguration UseCustomCryptographicAlgorithms(CngGcmAuthen
+            return UseCryptographicAlgorithmsCore(builder, options);
         }
 
-        private DataProtectionConfiguration UseCryptographicAlgorithmsCore(IInternalAuthenticatedEncryptionOptions options)
+        private static IDataProtectionBuilder UseCryptographicAlgorithmsCore(IDataProtectionBuilder builder, IInternalAuthenticatedEncryptionSettings options)
         {
             options.Validate(); // perform self-test
-            Use(DataProtectionServiceDescriptors.IAuthenticatedEncryptorConfiguration_FromOptions(options));
-            return this;
+            Use(builder.Services, DataProtectionServiceDescriptors.IAuthenticatedEncryptorConfiguration_FromOptions(options));
+            return builder;
         }
 
         /// <summary>
         /// Configures the data protection system to use the <see cref="EphemeralDataProtectionProvider"/>
         /// for data protection services.
         /// </summary>
+        /// <param name="builder">The <see cref="IDataProtectionBuilder"/>.</param>
         /// <returns>The 'this' instance.</returns>
         /// <remarks>
         /// If this option is used, payloads protected by the data protection system will
         /// be permanently undecipherable after the application exits.
         /// </remarks>
-        public DataProtectionConfiguration UseEphemeralDataProtectionProvider()
+        public static IDataProtectionBuilder UseEphemeralDataProtectionProvider(this IDataProtectionBuilder builder)
         {
-            Use(DataProtectionServiceDescriptors.IDataProtectionProvider_Ephemeral());
-            return this;
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            Use(builder.Services, DataProtectionServiceDescriptors.IDataProtectionProvider_Ephemeral());
+            return builder;
         }
 
         /*
          * UTILITY ISERVICECOLLECTION METHODS
          */
 
-        private void RemoveAllServicesOfType(Type serviceType)
+        private static void RemoveAllServicesOfType(IServiceCollection services, Type serviceType)
         {
             // We go backward since we're modifying the collection in-place.
-            for (int i = Services.Count - 1; i >= 0; i--)
+            for (int i = services.Count - 1; i >= 0; i--)
             {
-                if (Services[i]?.ServiceType == serviceType)
+                if (services[i]?.ServiceType == serviceType)
                 {
-                    Services.RemoveAt(i);
+                    services.RemoveAt(i);
                 }
             }
         }
 
-        private void Use(ServiceDescriptor descriptor)
+        private static void Use(IServiceCollection services, ServiceDescriptor descriptor)
         {
-            RemoveAllServicesOfType(descriptor.ServiceType);
-            Services.Add(descriptor);
+            RemoveAllServicesOfType(services, descriptor.ServiceType);
```

Comment:
Yes, but it doesn't affect public API and I'm trying not to change behavior. We know it works and we can fix it later if we want, given that it's private and just 10 lines of code, I don't feel it's worth fixing at this point.